flang/lib/Lower/OpenMP.cpp

```diff
@@ -55,8 +55,12 @@ static void createBodyOfOp(Op &op, Fortran::lower::FirOpBuilder &firOpBuilder,
   firOpBuilder.createBlock(&op.getRegion());
   auto &block = op.getRegion().back();
   firOpBuilder.setInsertionPointToStart(&block);
-  // Ensure the block is well-formed.
-  firOpBuilder.create<mlir::omp::TerminatorOp>(loc);
+  if (op.getOperationName() == "omp.wsloop") {
```

Comment:
Is there another way to do the comparison to avoid using a hard-coded string? Like `isa` or something like that.

Reply:
Yes, using `isa<FooOp>` is the preferred approach here.